hasjob/views/listing.py

```diff
@@ -972,3 +957,62 @@ def newjob():
     # 2. POST request from main page's Post a Job box
     # 3. POST request from this page, with errors
     return render_template('postjob.html', form=form, no_removelogo=True)
+
+
+@app.route('/<domain>/<hashid>/close', methods=('GET', 'POST'), defaults={'key': None}, subdomain='<subdomain>')
+@app.route('/<domain>/<hashid>/close', methods=('GET', 'POST'), defaults={'key': None})
+@app.route('/close/<hashid>', methods=('GET', 'POST'), defaults={'key': None, 'domain': None}, subdomain='<subdomain>')
+@app.route('/close/<hashid>/<key>', defaults={'domain': None}, methods=('GET', 'POST'), subdomain='<subdomain>')
+@app.route('/close/<hashid>', methods=('GET', 'POST'), defaults={'key': None, 'domain': None})
+@app.route('/close/<hashid>/<key>', defaults={'domain': None}, methods=('GET', 'POST'))
+def close(domain, hashid, key):
+    post = JobPost.query.filter_by(hashid=hashid).first_or_404()
+    if not ((key is None and g.user is not None and post.admin_is(g.user)) or (key == post.edit_key)):
+        abort(403)
+    if post.status not in POSTSTATUS.CLOSEABLE:
+        flash("Your job post can't be closed.", "info")
+        return redirect(url_for('index'), code=303)
```

Comment:
However, if the post can't be rendered because it's rejected (subject to #205), redirect to the domain or index.